tmo-monitor.py

```diff
@@ -134,79 +150,217 @@ def sha256url(self, val1, val2):
+      if site_meta['eNBID'] != config.connection['enbid']:
+        print('Not on eNB ID ' + str(config.connection['enbid']) + ', on ' + str(site_meta['eNBID']) + '.')
         reboot_requested = True
       else:
-        print('Camping on ' + band_4g + '.' + (' Not rebooting.' if not args.skip_reboot else ''))
-
-    if not args.skip_5g_bands:
+        print('eNB ID check passed, on ' + str(site_meta['eNBID']) + '.')
+
+    # Check for preferred bands regardless of reboot on band mismatch
+    if config.reboot['4G_band'] or config.reboot['5G_band']:
+      signal_info = tc_control.get_signal_info()
+
+      if config.connection['primary_band']:
+        primary_band = config.connection['primary_band']
+        band_4g = signal_info['cell_LTE_stats_cfg'][0]['stat']['Band']
+        if band_4g not in primary_band:
+          print('Not on ' + ('one of ' if len(primary_band) > 1 else '') + ', '.join(primary_band) + '.')
+          if config.reboot['4G_band']:
+            reboot_requested = True
+        else:
+          print('Camping on ' + band_4g + '.')
+
+      # 5G has a default value set (n41)
+      secondary_band = config.connection['secondary_band']
       band_5g = signal_info['cell_5G_stats_cfg'][0]['stat']['Band']
-      if band_5g not in args.secondary_band:
-        print('Not on ' + ('one of ' if len(args.secondary_band) > 1 else '') + ', '.join(args.secondary_band) + '.' + (' Reboot requested.' if not args.skip_reboot else ''))
-        reboot_requested = True
+      if band_5g not in secondary_band:
+        print('Not on ' + ('one of ' if len(secondary_band) > 1 else '') + ', '.join(secondary_band) + '.')
+        if config.reboot['5G_band']:
+          reboot_requested = True
       else:
-        print('Camping on ' + band_5g + '.' + (' Not rebooting.' if not args.skip_reboot else ''))
-
-
-  if not args.skip_ping and not reboot_requested and not tc_control.ping(args.ping_host, args.ping_count, args.ping_interval, args.interface):
-    print('Could not ping ' + args.ping_host + '.' + (' Reboot requested.' if not args.skip_reboot else ''))
-    reboot_requested = True
-
-  if args.skip_reboot and reboot_requested:
-    print('Skipping reboot.')
-  elif reboot_requested:
-    if args.reboot or tc_control.get_uptime() >= args.uptime:
-      print('Rebooting.')
-      tc_control.reboot()
+        print('Camping on ' + band_5g + '.')
+
+    # Check for successful ping
+    if config.reboot['ping']:
+      if not tc_control.ping(config.ping['ping_host'], config.ping['ping_count'], config.ping['ping_interval'], config.ping['interface']):
+        print('Could not ping ' + config.ping['ping_host'] + '.')
+        if config.reboot['ping']:
+          reboot_requested = True
+
+  if reboot_requested:
+    if config.skip_reboot:      
+      print('Not rebooting.')
     else:
-      print('Uptime threshold not met for reboot.')
+      print ('Reboot requested.')
+      if tc_control.get_uptime() >= config.reboot['uptime']:
```

Comment:
this is rewritten to no longer allow an explicit `--reboot` request to bypass the uptime requirement. practically speaking, this probably doesn't matter

Reply:
Easily fixed.